back/app/models/custom_page.rb

```diff
@@ -0,0 +1,104 @@
+#  top_info_section_enabled     :boolean          default(FALSE), not null
+#  top_info_section_multiloc    :jsonb            not null
+#  projects_enabled             :boolean          default(FALSE), not null
+#  projects_filter_type         :string
+#  events_widget_enabled        :boolean          default(FALSE), not null
+#  bottom_info_section_enabled  :boolean          default(FALSE), not null
+#  bottom_info_section_multiloc :jsonb            not null
+#  header_bg                    :string
+#  created_at                   :datetime         not null
+#  updated_at                   :datetime         not null
+#
+# Indexes
+#
+#  index_custom_pages_on_slug  (slug) UNIQUE
+#
+class CustomPage < ApplicationRecord
+  has_many :pins, as: :page, inverse_of: :page, dependent: :destroy
+  has_many :pinned_admin_publications, through: :pins, source: :admin_publication
+
+  has_many :text_images, as: :imageable, dependent: :destroy
+  accepts_nested_attributes_for :text_images
+
+  has_many :custom_page_files, -> { order(:ordering) }, dependent: :destroy, inverse_of: :custom_page
+
+  accepts_nested_attributes_for :pinned_admin_publications, allow_destroy: true
+
+  before_validation :sanitize_top_info_section_multiloc
+  before_validation :sanitize_bottom_info_section_multiloc
+
+  validates :title_multiloc, presence: true, multiloc: { presence: true }
+
+  validates :slug, presence: true, uniqueness: true
+
+  validates :banner_enabled, inclusion: [true, false] # Default is true on db table, perhaps should be false on db table?
+  validates :banner_layout, inclusion: %w[full_width_banner_layout two_column_layout two_row_layout]
+  validates :banner_overlay_color, css_color: true
+  validates :banner_overlay_opacity, numericality: { only_integer: true,
+                                                     in: [0..100],
+                                                     allow_nil: true }
+  validates :banner_header_multiloc, multiloc: true
+  validates :banner_subheader_multiloc, multiloc: true
+  validates :banner_cta_button_type, inclusion: %w[customized_button no_button]
+  with_options if: -> { banner_cta_button_type == 'customized_button' } do
+    validates :banner_cta_button_multiloc, presence: true, multiloc: { presence: true }
+    validates :banner_cta_button_url, presence: true, url: true
+  end
+
+  validates :top_info_section_enabled, inclusion: [true, false]
+  validates :top_info_section_multiloc, multiloc: { presence: false, html: true }
+
+  validates :projects_enabled, inclusion: [true, false]
+  with_options if: -> { projects_enabled == true } do
+    validates :projects_filter_type, presence: true, inclusion: %w[area topics]
+  end
+
+  validates :events_widget_enabled, inclusion: [true, false]
+
+  validates :bottom_info_section_enabled, inclusion: [true, false]
+  validates :bottom_info_section_multiloc, multiloc: { presence: false, html: true }
+
+  mount_base64_uploader :header_bg, HeaderBgUploader
```

Comment:
Maybe we should use a new `CustomPageBgUploader`?

Reply:
If this one works, I think it's fine. Less code - fewer bugs.